Chapter6/reliable_fsm.erl

```diff
@@ -0,0 +1,120 @@
+-module(reliable_fsm).
+-export([start/0, stop/0, un_hook/0, hook/0, other_un_hook/0, other_hook/0, call/1, get_call/1]).
+-export([idle/0, dial/0, wait_answer/1, ringing/1, connected/1]).
+
+start() -> register(cell, spawn(reliable_fsm, idle, [])).
+stop() 	-> cell ! stop.
+
+un_hook() 			-> msg(off_hook).
+hook() 				-> msg(on_hook).
+other_un_hook() 	-> msg(other_off_hook).
+other_hook() 		-> msg(other_on_hook).
+call(Number) 		-> msg({Number, calling}).
+get_call(Number) 	-> msg({Number, incoming}).
+
+
+% Exercise 6-2: A Reliable Mutex Semaphore
```

Comment:
I think that exercise 6.2 asks to improve mutex semaphore, that you [have](https://github.com/danieljharris/learning-erlang/blob/7c6aa19c6afdc9dbfa6839242439c8251d7310d5/Chapter5/mutex.erl) in module 5.
6.2 suggests to improve it by handling unreliable clients. 
> What happens if a process that currently holds the semaphore terminates prior to releasing it? Or what happens if a process waiting to execute is terminated due to an exit signal? By trapping exits and linking to the process that currently holds the semaphore, make your mutex semaphore reliable.

I don't see mutex is anyhow used here in `reliable_fsm`.